File: configlibrary/src/main/java/com/shmedo/core/enums/WorkModel.java
package com.shmedo.core.enums;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by adu on 2017/12/11.
 * 工作模式
 */
public enum WorkModel {
    INITIALZE(0), //初始化模式
    WORK(1), //工作模式
    DEBUG(2), //debug模式
    INFO(3);//info模式


    private int model;

    WorkModel(int i) {
        this.model = i;
    }

    public int toInt() {
        return model;
    }

    public static WorkModel valueOf(int model) {
        switch (model) {
            case 0:
                return INITIALZE;

            case 1:
                return WORK;

            case 2:
                return DEBUG;

            case 3:
                return INFO;

            default:
                return INITIALZE;
        }
    }

    public static boolean isValidMode(int value) {
        List<Integer> allModes = new ArrayList<>();
        for (WorkModel model : WorkModel.values()) {
            allModes.add(model.toInt());
        }

        return allModes.contains(value);
    }
}

File: lib_device_base/src/main/java/com/shmedo/lib/device/base/md_cmd/enums/WorkModel.kt
package com.shmedo.lib.device.base.md_cmd.enums

/**
 * Created by adu on 2017/12/11.
 * 工作模式
 */
enum class WorkModel(//info模式
    private val model: Int
) {
    INITIALZE(0),

    //初始化模式
    WORK(1),

    //工作模式
    DEBUG(2),

    //debug模式
    INFO(3);

    fun toInt(): Int {
        return model
    }

    companion object {
        fun value(model: Int): WorkModel {
            return when (model) {
                0 -> INITIALZE
                1 -> WORK
                2 -> DEBUG
                3 -> INFO
                else -> INITIALZE
            }
        }

        fun isValidMode(value: Int): Boolean {
            val allModes: MutableList<Int> = ArrayList()
            for (model in entries) {
                allModes.add(model.toInt())
            }
            return allModes.contains(value)
        }
    }
}
